File: 424-longest-repeating-character-replacement/longest-repeating-character-replacement.cpp
class Solution {
public:
    int characterReplacement(string s, int k) {
        
        unordered_map<char,int> mp;
        int max_freq=0;
        int l=0,r=0;
        int max_len=0;
        while(r<s.length())
        {
            mp[s[r]]++;
            max_freq= max(max_freq,mp[s[r]]);

            if((r-l+1)-max_freq > k)
            {
                mp[s[l]]--;
                l++;
            }
            else{
                max_len = max(max_len,r-l+1);
            }
            r++;
        }
        return max_len;
    }
};

File: 424-longest-repeating-character-replacement/longest-repeating-character-replacement.py
class Solution:
    def characterReplacement(self, s: str, k: int) -> int:
        my_dict={}
        max_freq=0
        l,r=0,0
        max_len=0
        while r<len(s):
            if s[r] in my_dict:
                my_dict[s[r]]+=1
            else:
                my_dict[s[r]]=1
            
            max_freq = max(max_freq,my_dict[s[r]])

            if (r-l+1)-max_freq<=k:
                max_len = max(max_len,r-l+1)
            
            else:
                my_dict[s[l]]-=1
                l+=1

            r+=1

        return max_len
